### DISCUSSION TITLE
Unknown column 'cert_dxcc' in 'order clause'

### DISCUSSION THREAD

**Discussion by dh5ym** [magicbug/Cloudlog#2658] [General]
I still run an old 1.7 installation on my webserver. Yesterday i tried to upgrade to the latest version. Unfortunately it did not work because my PHP is apparently too old. Therefore i reverted to the backup of the old installation. As far as i can see everything works except the lotw support. The page does not show up and instead a white page with the link to upload the certificate.
I enabled the development mode and thats the error message:

`
A Database Error Occurred

Error Number: 1054

Unknown column 'cert_dxcc' in 'order clause'

SELECT * FROM `lotw_certs` WHERE `user_id` = '6' ORDER BY `cert_dxcc` ASC

Filename: models/LotwCert.php

Line Number: 17
`

Does anybody have an idea how to fix this?

73
Mario

**Comment by HB9HIL** [comment]
Hello

How did you backup your installation? And how did you revert to that backup after the failed installation?

73

**Comment by phl0** [comment]
Has the database been backed up and restored as well?

**Comment by dh5ym** [comment]
All i did was to make a copy of the old folder. This copy i renamed back after i noticed that the update wont work as expected since i need to upgrade the whole Linux machine.
I did not do a backup of the background database. So maybe i messed up some things.
The good news is that appart from the lotw support i dont find problems so far. I could live with adding the keys again but right now i dont really have a clue how to proceed.

**Comment by phl0** [comment]
Which PHP version is running now? The code should still be able to cope with 7.4.
Maybe you can update to 7.4 only?

**Comment by dh5ym** [comment]
Its a 7.2... i will have a look. 
Getting the lotw support in the old one back to work would be fine as well. Is my understanding correct that my attempt to run the new version changed things in the database which may not be compatible with the old version of cloudlog?

**Comment by HB9HIL** [comment]
For the Database Error  on the LOTW Page you CAN try following:

login as root with `sudo su` and your user password

First create a backup of your current databases:

`mysqldump --all-databases > full_backup.sql`  


then go into the mysql command line with `mysql`

Type in 

```
USE cloudlog;
ALTER TABLE lotw_certs 
RENAME COLUMN cert_dxcc_id TO cert_dxcc;
QUIT;
```
// Make sure the name of your database (in my case its cloudlog). If you dont know check `application/config/database.php` for the parameter `database` (around Line 82)

Now try again the LoTW Page.

**Comment by HB9HIL** [comment]
But anyway I would recommend to export your complete ADIF, note your setting  and install a new cloudlog on a new setup Debian. This will be much easier to update everything on it's own. 1.9 is really really old and this would be a huge version step. 

I recommend on the new Cloudlog Install to set up the cronjob to update each night. 


Would that be an option? If you need help you can write me a german email. Mail is on QRZ.com

**Comment by dh5ym** [comment]
Hi, 

thanks for the recommendation.
I guess  thats the only good way to go. But it will take me quite some time. There are other requirements on this server that prevent me to do a fresh installation now.
I will backup all my log data and try to fix the lotw upload in the old installation.

73
Mario

**Comment by dh5ym** [comment]
Its a 7.2... i will have a look. 
Getting the lotw support in the old one back to work would be fine as well. Is my understanding correct that my attempt to run the new version changed things in the database which may not be compatible with the old version of cloudlog?

**Comment by dh5ym** [comment]
I managed to change the column name (my mariadb version did not yet support the RENAME command). The lotw page is now working again. But there seem also other things changed in other tables that then prevent log entries from beeing uploaded (download seems to work). I dont want to mess up my log data and therefore i plan to do a new installation on a current system and then move all logbooks to there manually.
Thanks for your help.
Mario
